src/states_screens/race_gui_overworld.cpp

```diff
@@ -443,6 +443,15 @@ void RaceGUIOverworld::drawGlobalMiniMap()
         const ChallengeData* challenge = unlock_manager->getChallengeData(challenges[n].m_challenge_id);
         const unsigned int val = challenge->getNumTrophies();
         bool unlocked = (PlayerManager::getCurrentPlayer()->getPoints() >= val);
+	if (challenges[n].m_challenge_id == "fortmagma") {
+		// For each track, check whether the easy difficulty has been completed ; fortmagma will not affect our decision (`n == m`) ; tutorial is ignored because it has no completion level
+		for (unsigned int m = 0; unlocked && m < challenges.size(); m++) {
+			if (challenges[m].m_challenge_id == "tutorial") continue;
+			unlocked = unlocked &
+				(PlayerManager::getCurrentPlayer()->getChallengeStatus(challenges[m].m_challenge_id)->isSolved(RaceManager::DIFFICULTY_EASY) || n == m);
```

Comment:
Does that also work when you solved a challenge on higher difficulty ?
I haven't checked where the isSolved method is implemented.

Reply:
It did (and actually, I solved all challenges in difficult first, which is what made me notice the discrepancy I'm fixing here). I checked both low and high difficulty, and saw other uses of `isSolved` in the already existing code. If `RaceManager::DIFFICULTY_HIGH` or `RaceManager::DIFFICULTY_MEDIUM` are completed for a track, `isSolved` will also return `true` if we check the lower difficulty (which makes sense since both medium and easy levels are marked as complete if you do the expert mode first).

I noticed though that I could use `isSolvedAtAnyDifficulty`, which, perhaps, is less confusing. Should I change it?

Reply:
Neat, I suppose it is not less confusing for someone used to the base code, but I find it easier to read without knowing much from the rest of the code.

Reply:
Then it shall be pushed in a minute or so (I had it coded and tested since I posted my answer).
Look out for the new diff right after.